tar/bsdtar.c

```diff
@@ -804,8 +804,15 @@ main(int argc, char **argv)
 		    "Must specify one of -c, -r, -t, -u, -x");
 
 	/* Check boolean options only permitted in certain modes. */
-	if (bsdtar->flags & OPTFLAG_AUTO_COMPRESS)
-		only_mode(bsdtar, "-a", "c");
+	if (bsdtar->flags & OPTFLAG_AUTO_COMPRESS) {
+		only_mode(bsdtar, "-a", "cx");
+		if (strchr("x", bsdtar->mode) != NULL) {
+			bsdtar->flags &= ~OPTFLAG_AUTO_COMPRESS;
+			lafe_warnc(0,
+			    "Option -a makes no sense in mode -x. "
+			    "Allowing SOLELY for GNU tar compatibility");
```

Comment:
I would just say "Ignoring -a in mode -x".
In particular, the reference to GNU tar isn't really helpful.